api/loaders/timeline.js

```diff
@@ -0,0 +1,108 @@
+import _ from 'lodash';
+
+export const MAX_PAGE_SIZE = 1000;
+export const DEFAULT_PAGE_SIZE = 100;
+
+function parseStreamEntry(data) {
+  if (!data || data.length < 2) {
+    return null;
+  }
+
+  let k = data[0];
+  let v = data[1][1];
+
+  return [k, v];
+}
+
+/**
+ * getTimelineInfo: returns attributes of the specified timeline type to inform pagination and auto-scroll controls
+ *
+ * @param timeline
+ * @returns {Promise<{last: *, length: *, timeline: *, first: *, ts: *}>}
+ */
```

Comment:
this class lets us provide timelines in a somewhat-generic way for blocks, transactions and program transactions